client/src/app/plan.service.ts

```diff
@@ -99,6 +99,57 @@ export class PlanService {
     return this.user.pipe(map(mapCallback));
   }
 
+  /**
+   * Adds a new semester into years chronologically. If term already exists, nothing is done.
+   * @param term The term in string form
+   * @param yearIndex  the Index of the year that the semester will be added to
+   * @param years The years array
+   */
+  addNewSemester(term: string, yearIndex: number, years: Array<any>): void {
```

Comment:
This function handles changing the plan variable in s-degree-plan-editor to add a new semester.